### PR TITLE
Happy path API Provider error handling

### PR DESCRIPTION
**Pull request by mmuzikar**
//test: `@api-provider`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/apiprovider/
  ApiProviderFlowSteps.java:37:49: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/utils/
  ByUtils.java:7:51: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/utils/
  ByUtils.java:11:63: '{' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Validate all credentials 
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu 
Connection - CRUD | Kebab menu edit and view 
API Provider Integration | Create API provider from url spec todo-app 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider GET non-existent 
API Provider Integration | API Provider GET collection 
API Provider Integration | API Provider GET emptycollection 
API Provider Integration | API Provider POST new 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider POST collection 
API Provider Integration | API Provider PUT 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider not visible in connections 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - edit unimplemented operation 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | Clicking Go To Operation List does not discard progress in API Provider 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type 
API Provider Integration | API Provider start and stop connection cannot be deleted 
API Provider Integration | Base path in API Provider URL 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | Check that API Provider shows the number of flow instead of steps for random flow 
API Provider Integration | API Provider error handling tests 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu edit and view 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider GET non-existent 
API Provider Integration | API Provider GET collection 
API Provider Integration | API Provider GET emptycollection 
API Provider Integration | API Provider POST new 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider POST collection 
API Provider Integration | API Provider PUT 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - edit unimplemented operation 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | API Provider error handling tests 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET non-existent 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider Edit OpenAPI - edit unimplemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | API Provider error handling tests 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - Database | Smoke - Periodic invocation to Insert 
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu edit and view 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider GET non-existent 
API Provider Integration | API Provider GET collection 
API Provider Integration | API Provider GET emptycollection 
API Provider Integration | API Provider POST new 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider POST collection 
API Provider Integration | API Provider PUT 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - edit unimplemented operation 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | API Provider operation with simple return type 
API Provider Integration | Base path in API Provider URL 
API Provider Integration | Conditional flows used in API Provider 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider GET non-existent 
API Provider Integration | API Provider GET collection 
API Provider Integration | API Provider GET emptycollection 
API Provider Integration | API Provider POST new 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider POST collection 
API Provider Integration | API Provider PUT 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider not visible in connections 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - edit unimplemented operation 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | Clicking Go To Operation List does not discard progress in API Provider 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type 
API Provider Integration | API Provider start and stop connection cannot be deleted 
API Provider Integration | Base path in API Provider URL 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | Check that API Provider shows the number of flow instead of steps for random flow 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete unimplemented operation 
API Provider Integration | Publishing an integration without any operation should not be possible 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | Clicking Go To Operation List does not discard progress in API Provider 
API Provider Integration | Clicking back button from Apicurio should work consistently in API Provider 
API Provider Integration | API Provider operation with simple return type 
API Provider Integration | API Provider start and stop connection cannot be deleted 
API Provider Integration | Base path in API Provider URL 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | Check that API Provider shows the number of flow instead of steps for random flow 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
I implemented the get-non-existing test as a happy path for testing error handling, the test covers most of the errors that are defined in Syndesis surprisingly, I will add more test cases as more errors get added.

File: ui-tests/src/test/resources/features/integrations/api-provider.feature [hunk 1]
```diff
@@ -176,17 +177,27 @@ Feature: API Provider Integration
       | id        | body.id        |
       | completed | body.completed |
       | task      | body.task      |
-    And sleep for jenkins delay or "2" seconds
     And check "Done" button is "visible"
     And click on the "Done" button
 
+    And edit integration step on position 5
+    And map Api Provider errors
+      | Server Error         | 404 |
```
Comment (asmigala) [inline ui-tests/src/test/resources/features/integrations/api-provider.feature]: I'd suggest setting a different error code here to see that it does not return 404 for everything now :D

Comment (avano) [reply ui-tests/src/test/resources/features/integrations/api-provider.feature]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418 :grin:

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | Check that API Provider shows the number of flow instead of steps for random flow 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
@avano Except for the last one yes those are expected

**Comment by avano** [comment]
@mmuzikar would it be possible to annotate failing tests with issue numbers? also will you fix the failing one in this PR?

**Comment by mmuzikar** [comment]
@avano The failing tests are annotated, but they are closed on github due to the migration to JIRA, I will create a separate pull request for `Check that API Provider shows the number of flow instead of steps for random flow`, this can be verified manually quite easily in the mean time.

**Comment by mmuzikar** [comment]
I will add annotations for respective JIRAs, didn't see that we have tagging for JIRA issues already.

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Kebab menu 
Connection - CRUD | Kebab menu edit and view 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.json 
API Provider Integration | Create API provider from file spec swagger/connectors/todo.swagger.yaml 
API Provider Integration | Create from scratch 
API Provider Integration | API Provider GET single 
API Provider Integration | API Provider GET non-existent 
API Provider Integration | API Provider GET collection 
API Provider Integration | API Provider GET emptycollection 
API Provider Integration | API Provider POST new 
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider POST collection 
API Provider Integration | API Provider PUT 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider export roundtrip 
API Provider Integration | API Provider Edit OpenAPI - add operation 
API Provider Integration | API Provider Edit OpenAPI - edit unimplemented operation 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | API Provider Edit OpenAPI - delete implemented operation 
API Provider Integration | Publishing an integration with all operations deleted should not be possible 
API Provider Integration | API Provider operation with simple return type 
API Provider Integration | Base path in API Provider URL 
API Provider Integration | Conditional flows used in API Provider 
API Provider Integration | Check that API Provider shows the number of flow instead of steps for random flow 
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
API Provider Integration | API Provider POST existing 
API Provider Integration | API Provider DELETE 
API Provider Integration | API Provider Edit OpenAPI - edit implemented operation 
API Provider Integration | Conditional flows used in API Provider 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
@asmigala This is also ready to merge on my side, I will make another PRs to fix the failing tests

**Pull request merged**